types/parse/index.d.ts

```diff
@@ -657,15 +668,23 @@ namespace Parse {
             X extends Extract<keyof U['attributes'], string>>(key: K, queryKey: X, query: Query<U>): this;
         doesNotMatchQuery<U extends Object, K extends keyof T['attributes']>(key: K, query: Query<U>): this;
         distinct<K extends keyof T['attributes'], V = T['attributes'][K]>(key: K): Promise<V>;
-        each(callback: (obj: T) => PromiseLike<void> | void, options?: Query.EachOptions): Promise<void>;
+        eachBatch(callback: (objs: T[]) => Promise<any>, options?: Query.BatchOptions): Promise<void>;
+        each(callback: (obj: T) => any, options?: Query.BatchOptions): Promise<void>;
+        hint(value: string | object): this;
+        explain(explain: boolean): this;
+        map(callback: (currentObject: T, index: number, query: Query) => any, options?: Query.BatchOptions): Promise<any[]>;
+        reduce(callback: (accumulator: any, currentObject: T, index: number) => any, initialValue: any, options?: Query.BatchOptions): Promise<any[]>;
+        filter(callback: (currentObject: T, index: number, query: Query) => boolean, options?: Query.BatchOptions): Promise<T[]>;
```

Comment:
The callback can also return a Promise https://parseplatform.org/Parse-SDK-JS/api/master/Parse.Query.html#filter

```suggestion
        filter(callback: (currentObject: T, index: number, query: Query) => PromiseLike<boolean> | boolean, options?: Query.BatchOptions): Promise<T[]>;
```